runtime/compiler/control/CompilationThread.cpp

```diff
@@ -6977,7 +6979,46 @@ TR::CompilationInfoPerThreadBase::preCompilationTasks(J9VMThread * vmThread,
       if ((_compInfo.getPersistentInfo()->getJITaaSMode() == CLIENT_MODE) &&
          !shouldPerformLocalComp(entry))
          {
-         entry->setRemoteCompReq();
+         JITaaS::J9ClientStream *client = getClientStream();
+         if (!client)
+            {
+            if (JITaaS::J9ClientStream::_serverReachable || JITaaS::J9ClientStream::_numLocalCompilations >= 50)
```

Comment:
Using the number of compilations as a counter/timer is convenient, but it might cause a lot of variation and fluctuation in the long run because "50 compilations" might take 1 second or 100 seconds, depending on a lot of variables. Is it possible to use wall clock time instead? End users who need to tweak this sort of thing will probably have an easier time with wall clocks than the number of compilations.

Reply:
yea i agree, I will switch to a timer